controllers/resources/bentorequest_controller.go

```diff
@@ -1756,32 +1767,54 @@ if [ -f "{{.ModelDirPath}}/.exists" ]; then
 	exit 0
 fi
 
+if [ -n "{{.HfToken}}" ]; then
+	echo "HF_TOKEN is set, using HF_TOKEN to download model..."
+	export HF_TOKEN="{{.HfToken}}"
+fi
+
 mkdir -p {{.ModelDirPath}}
 url="{{.ModelDownloadURL}}"
-echo "Downloading model {{.ModelRepositoryName}}:{{.ModelVersion}} to /tmp/downloaded.tar..."
-if [[ ${url} == s3://* ]]; then
-	echo "Downloading from s3..."
-	aws s3 cp ${url} /tmp/downloaded.tar
-elif [[ ${url} == gs://* ]]; then
-    echo "Downloading from GCS..."
-    gsutil cp ${url} /tmp/downloaded.tar
+
+if [[ ${url} == hf://* ]]; then
+	mkdir -p /tmp/model
+	hf_url="${url:5}"
+	model_id=$(echo "${hf_url}" | cut -d '@' -f 1)
+	endpoint=$(echo "${hf_url}" | cut -d '@' -f 2)
+	revision=$(echo "{{.ModelTag}}" | cut -d ':' -f 2)
+	echo "Downloading model ${model_id} (endpoint=${endpoint}, revision=${revision}) from Huggingface..."
+	export HF_ENDPOINT=${endpoint}
+	huggingface-cli download ${model_id} --revision ${revision} --local-dir /tmp/model
+	echo "Moving model to {{.ModelDirPath}}..."
+	mv /tmp/model/* {{.ModelDirPath}}
+	rm -rf /tmp/model
 else
-	curl --fail -L -H "{{.ModelDownloadHeader}}" ${url} --output /tmp/downloaded.tar --progress-bar
+	echo "Downloading model {{.ModelRepositoryName}}:{{.ModelVersion}} to /tmp/downloaded.tar..."
+	if [[ ${url} == s3://* ]]; then
+		echo "Downloading from s3..."
+		aws s3 cp ${url} /tmp/downloaded.tar
```

Comment:
‌‌‌‌‌‌‌Is this acceptable? This way, there's no need to clean up /tmp/downloaded.tar, it reduces two disk I/O operations, and it decreases the writing to the container's writable layer, thus reducing the usage of ephemeral storage.

```bash
aws s3 cp {url} - | tar -xv -C {{.ModelDirPath}}
```